### PR TITLE
Refactoring simplify_inequality_*

### PR DESCRIPTION
**Pull request by kroening**
- [X] Each commit message has a non-empty body, explaining why the change was made.
- [X] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- n/a The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [X] My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: src/util/simplify_expr_int.cpp [hunk 1]
```diff
@@ -1478,19 +1372,147 @@ bool simplify_exprt::simplify_inequality(exprt &expr)
...
+  if(tmp0.type().id() == ID_c_enum_tag)
+    tmp0.type() = ns.follow_tag(to_c_enum_tag_type(tmp0.type()));
+
+  if(tmp1.type().id() == ID_c_enum_tag)
+    tmp1.type() = ns.follow_tag(to_c_enum_tag_type(tmp1.type()));
+
+  const auto &tmp0_const = to_constant_expr(tmp0);
+  const auto &tmp1_const = to_constant_expr(tmp1);
+
+  // are _both_ constant?
```
Comment (smowton) [inline src/util/simplify_expr_int.cpp]: Now this is moved, comment no longer needed

**Review by smowton — APPROVED** [review]

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: d03c701183333a17c59bb7b84110b67f4050f635).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/113432166

File: src/util/simplify_expr_int.cpp (hunk 1 above)
Comment (kroening) [reply src/util/simplify_expr_int.cpp]: done

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 5b2674306a675bc11d1223a3149a063a4617e48b).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/113438375

**Pull request merged**